domain/src/main/java/org/oppia/domain/util/StateRetriever.kt

```diff
@@ -477,4 +488,54 @@ class StateRetriever @Inject constructor(
     }
     return StringList.newBuilder().addAllHtml(list).build()
   }
+
+  private fun parseImageWithRegions(jsonObject: JSONObject): ImageWithRegions {
+    return ImageWithRegions.newBuilder()
+      .addAllLabelRegions(parseJsonToLabeledRegionsList(jsonObject.getJSONArray("labeledRegions")))
+      .setImagePath(jsonObject.getString("imagePath"))
+      .build()
+  }
+
+  private fun parseJsonToLabeledRegionsList(jsonArray: JSONArray): List<LabeledRegion> {
+    val list: MutableList<LabeledRegion> = ArrayList()
+    for (i in 0 until jsonArray.length()) {
+      list.add(parseLabeledRegion(jsonArray.getJSONObject(i)))
+    }
+    return list
+  }
+
+  private fun parseLabeledRegion(jsonObject: JSONObject): LabeledRegion {
+    return LabeledRegion.newBuilder()
+      .setLabel(jsonObject.getString("label"))
+      .setRegion(parseRegion(jsonObject.getJSONObject("region")))
+      .setRegionDescription(jsonObject.getString("regionDescription"))
```

Comment:
Similar to earlier comment: this isn't supported yet, right?

Reply:
done